### PR TITLE
Service Mesh Interface (SMI) proposal

### LINKED ISSUE #337: Service Mesh Interface (SMI) proposal
We would like to propose [Service Mesh Interface (SMI)](https://smi-spec.io/) specification for donation as a sandbox project in the CNCF.

The SMI proposal was presented during the CNCF sig-network, January 16, 2020. The proposal deck can be found [here](https://docs.google.com/presentation/d/1Pn6PvdaCnKf-MRhATWDFgKGmM7Z6tycumb07xYSev5U/edit?ts=5e179825#slide=id.g4054009e6b_5_0).

Please find associated proposal PR - #336

### PR DESCRIPTION
**Pull request by lachie83**
fixes #337

### CODE REVIEW

**Comment by caniszczyk** [comment]
@mattklein123 agrees to sponsor as a TOC member for the sandbox

File: proposals/smi.adoc [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+SMI is complementary to existing CNCF projects like LinkerD; it encourages adoption of service mesh. SMI is the only cross-industry effort to make service meshes interoperable. For example, LinkerD is one example of a service mesh that implements and uses SMI.
+
+*Sponsor from TOC (sponsor helps mentor projects):*
+TBD
+
+*Preferred maturity level (see CNCF Graduation Criteria):*
+Sandbox
+
+*License (charter dictates Apache 2 by default):* 
+The current license is OWF Contributor License Agreement 1.0 - Copyright and Patent. - https://github.com/deislabs/smi-spec/blob/master/CONTRIBUTING.md#specifications-subject-to-the-open-web-foundation-agreements
```
Comment (edwarnicke) [inline proposals/smi.adoc]: Could we get a little background on the license choice?

File: proposals/smi.adoc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+Service Mesh Interface
+
+*Project Description:*
+Service Mesh Interface is an open source specification for interoperable service meshes on Kubernetes. Launched in May 2019, SMI brings the ecosystem together with a standard which enables innovation by providing a common abstraction that anyone can build against.
+
+*Statement on alignment with CNCF charter mission:*
+The Service Mesh Interface spec is well-aligned with the CNCF mission to make cloud native computing ubiquitous. Interoperability creates a virtuous cycle which increases adoption. SMI encourages an open source, vendor-neutral service mesh ecosystem by empowering end users to build upon ecosystem tools without needing to re-implement the underlying substrate.
+
+*Comparison with similar projects (inside or outside the CNCF), including what differentiates this project:*
+SMI is complementary to existing CNCF projects like LinkerD; it encourages adoption of service mesh. SMI is the only cross-industry effort to make service meshes interoperable. For example, LinkerD is one example of a service mesh that implements and uses SMI.
```
Comment (fkautz) [inline proposals/smi.adoc]: Are there any other examples besides linkerd? I recall there being a large cabal of companies who support it.

Comment (sdake) [inline proposals/smi.adoc]: `SMI is the only cross-industry effort to make service meshes interoperable` isn't accurate. There is also [Hamlet](https://octo.vmware.com/project-hamlet-secure-multi-vendor-multi-mesh-federation-open-source/). From the `Closing Remarks` section of that announcement:

```
Closing Remarks

Through Hamlet and the collaborative effort of the teams involved within Istio, Google Cloud’s Anthos, Hashicorp Consul, Pivotal, and VMware NSX Service Mesh, service meshes can now interoperate, enabling customers to discover and securely communicate services across these meshes.
```

Comment (caniszczyk) [reply proposals/smi.adoc]: Hamlet hasn't had activity in the last ~4 months so maybe it's a dead project now? Prior art should still be mentioned though

Comment (sdake) [reply proposals/smi.adoc]: @sergiopozoh can add more information on the maintenance status of Hamlet. My understanding is Hamlet continues to be a running project.

Cheers
-steve

**Comment by sergiopozoh** [comment]
Thank you. @sdake @caniszczyk yes, Hamlet is a living project. As @sdake described, Hamlet aims to interoperate service discovery and related topics in single-vendor and multi-vendor meshes, which includes the automated configuration management of the involved meshes. Hamlet has been presented in the Istio Environments WG and we discussed an integration with Istio that would be eventually contributed.

My personal opinion is that Hamlet and SMI are compatible and complementary. Hamet does not make assumptions of the underlying service mesh. Which means that either vendor-native APIs or SMI APIs (if supported) can be used to configure each federated mesh.

In addition to that, I imagine that since SMI imagines a multi-vendor world, it would at some point need these multi-vendor meshes to interoperate (service discovery, service to service secure communication, etc.).

**Comment by caniszczyk** [comment]
Any other @cncf/toc members open to sponsoring SMI?

File: proposals/smi.adoc (hunk 1 above)
Comment (lachie83) [reply proposals/smi.adoc]: Disclaimer - IANAL.

This license was selected as it is generally used for industry specifications and covers the details for releasing and versioning specifications. This licensing is also adopted by the OCI as referenced in their charter documentations. Here is some reference material.
* [OCI v1.0.0 release notice](https://www.opencontainers.org/release-notices/v1-0-0)
* [OCI Governance docs](https://www.opencontainers.org/about/governance)
* [OWF CLA Copyright and Patent](http://www.openwebfoundation.org/legal/the-owf-1-0-agreements/owf-contributor-license-agreement-1-0---copyright-and-patent)

File: proposals/smi.adoc (hunk 2 above)
Comment (lachie83) [reply proposals/smi.adoc]: Added callout to Hamlet in **Comparison with similar projects** section

**Comment by michelleN** [comment]
> Any other @cncf/toc members open to sponsoring SMI?

I'm happy to sponsor

**Comment by michelleN** [comment]
Looks like @mattklein123 is also willing to sponsor https://github.com/cncf/toc/issues/337#issuecomment-575353837

**Comment by quinton-hoole** [comment]
nit: Which of the several implementations is considered to be "the reference implementation".  We've previously had a policy of not admitting specs without reference implementations.  

My apologies if the answer is buried in there somewhere, but I didn't see it during my skim-read.

File: proposals/smi.adoc (hunk 1 above)
Comment (edwarnicke) [reply proposals/smi.adoc]: Cool!  IANAL either... but I've walked enough things through legal to want to make sure we aren't making up our own licenses and are adopting best practices from others :)

**Comment by lachie83** [comment]
> nit: Which of the several implementations is considered to be "the reference implementation". We've previously had a policy of not admitting specs without reference implementations.
> 
> My apologies if the answer is buried in there somewhere, but I didn't see it during my skim-read.

Given that SMI has a stated non-goal of: Implement a service mesh offered by the SMI project. Here's what I would consider reference implementations:
* https://github.com/deislabs/smi-sdk-go
* https://github.com/deislabs/smi-adapter-istio
* https://github.com/deislabs/smi-metrics

In addition, the following service meshes natively implement SMI:
* Linkerd: linkerd.io
* Rio: rio.io
* Maesh: mae.sh
* SuperGloo: supergloo.solo.io

File: proposals/smi.adoc (hunk 1 above)
Comment (lachie83) [reply proposals/smi.adoc]: Appreciate the due-diligence and I agree with your statement about NOT "making" up our own licenses but rather adopting best practices.

**Comment by sergiopozoh** [comment]
@lachie83 is SuperGloo a Service Mesh? From their own definition, it doesn't seem to: "SuperGloo, an open-source project to manage and orchestrate service meshes at scale. SuperGloo is an opinionated abstraction layer that will simplify the installation, management, and operation of your service mesh, whether you use (or plan to use) a single mesh or multiple mesh technologies, on-site, in the cloud, or on any topology that best fits you."

**Comment by lachie83** [comment]
@sergiopozoh appreciate the feedback. I've updated my original comment to hopefully make this clearer.

**Comment by sergiopozoh** [comment]
@lachie83 that's ok. Would you like to add Consul? I believe they support part of the SMI spec cc @nicholasjackson

**Comment by lachie83** [comment]
@sergiopozoh - At this point it's probably worth mentioning that you can refer to [slide 8](https://docs.google.com/presentation/d/1Pn6PvdaCnKf-MRhATWDFgKGmM7Z6tycumb07xYSev5U/edit?ts=5e179825#slide=id.g4054009e6b_5_26) of the proposal deck. Consul is there. Feel free to let me know if we've missed any other implementations.

**Comment by sergiopozoh** [comment]
Thank you for pointing me to the deck! Appreciated!

**Comment by brewer531** [comment]
> Any other @cncf/toc members open to sponsoring SMI?
> 
> I'm happy to sponsor

I am happy to sponsor as well.

**Comment by amye** [comment]
@leecalcote: Checking in on SIG-Network's progress on review

**Comment by lachie83** [comment]
@amye - It has been brought to my attention that the CNCF doesn't currently accept any licensing other than Apache v2. We are willing to have the specification relicensed to Apache v2 as part of the process to donate this to the CNCF.

**Comment by leecalcote** [comment]
@lachie83 very good. I'm noting this change. Thank you for this update.

**Comment by lachie83** [comment]
Hey @leecalcote, can you kindly confirm the next steps in the process and provide an approximate timeline?  We would like to complete this donation and are ready to assist with anything that needs to happen to finish it.

**Comment by leecalcote** [comment]
All, for reference, please see SIG-Network's review and recommendation of this project proposal - https://github.com/cncf/sig-network/blob/master/recommendations/smi.md. We consider SMI a great fit for Sandbox and a very welcomed addition. SMI advances the adoption and interoperability of services meshes as a prominent, and soon to be ubiquitious, layer of cloud native infrastructure. 

SIG-Network recommends SMI for Sandbox. 👍

**Comment by michelleN** [comment]
@amye -- this proposal has gone through all the steps in the sandbox process and has the necessary sponsors, so I'm marking them as done. I'll leave it to you and @caniszczyk from here to get them over the line with any final steps.

**Comment by caniszczyk** [comment]
@michelleN it was on my TODO list for today, you're official now and will be added to devstats shortly, we need to work on migrating the SMI repos over to github.com/servicemeshinterface

https://github.com/cncf/artwork/blob/master/examples/sandbox.md#service-mesh-interface-logos

**Comment by tomkerkhove** [comment]
Awesome! 🎉

**Comment by Pothulapati** [comment]
This is really exciting! :fireworks:

**Pull request merged**